pipeline/trusted-resources/cmd/sign/sign_test.go

```diff
@@ -91,94 +77,68 @@ var (
 	}
 )
 
-func TestSign_Taskrun(t *testing.T) {
+func TestSignTaskrun(t *testing.T) {
 	ctx := context.Background()
 
 	sv, err := trustedtask.GetSignerVerifier(password)
 	if err != nil {
 		t.Fatalf("error get signerverifier: %v", err)
 	}
 
-	tr := &v1beta1.TaskRun{
-		TypeMeta:   trTypeMeta,
-		ObjectMeta: trObjectMeta,
-		Spec: v1beta1.TaskRunSpec{
-			TaskSpec: &ts.Spec,
-		},
-	}
+	tr := getTaskRun()
 
 	var writer bytes.Buffer
 
-	if err := Sign(ctx, tr, sv, &writer); err != nil {
+	if err := SignTaskRun(ctx, tr, sv, &writer); err != nil {
 		t.Fatalf("Sign() get err %v", err)
 	}
 
 	signed := writer.Bytes()
-	tr, signature := unmarshal(t, signed)
+	tr, signature := unmarshalTaskRun(t, signed)
 
-	if err := trustedtask.VerifyInterface(ctx, tr.Spec.TaskSpec, sv, signature); err != nil {
+	if err := trustedtask.VerifyInterface(ctx, tr, sv, signature); err != nil {
 		t.Fatalf("VerifyTaskOCIBundle get error: %v", err)
 	}
 
 }
 
-func TestSign_OCIBundle(t *testing.T) {
-	ctx := context.Background()
 
-	sv, err := trustedtask.GetSignerVerifier(password)
-	if err != nil {
-		t.Fatalf("error get signerverifier: %v", err)
+func unmarshalTaskRun(t *testing.T, buf []byte) (*v1beta1.TaskRun, []byte) {
+	tr := &v1beta1.TaskRun{}
+	if err := yaml.Unmarshal(buf, &tr); err != nil {
+		t.Fatalf("error unmarshalling buffer: %v", err)
 	}
 
-	k8sclient := fakek8s.NewSimpleClientset(sa)
-
-	// Create registry server
-	s := httptest.NewServer(registry.New())
-	defer s.Close()
-	u, err := url.Parse(s.URL)
+	signature, err := base64.StdEncoding.DecodeString(tr.Annotations[trustedtask.SignatureAnnotation])
 	if err != nil {
-		t.Fatalf("error parsing url: %v", err)
+		t.Fatalf("error decoding signature: %v", err)
 	}
 
-	// Push OCI bundle
-	pushOCIImage(t, u, ts)
+	delete(tr.ObjectMeta.Annotations, trustedtask.SignatureAnnotation)
```

Comment:
copied from the other PR - why do we need this delete here? wouldn't we want to keep the TaskRun unaltered?

Reply:
it is used before verification, so we need to remove the signature from the taskrun then we can verify the whole taskrun

Reply:
This might be a sign that we should change the API a bit -
It's a bit weird that we're marshaling a TaskRun struct to unmarshal it to marshal it again. 🤔  This probably means we should refactor the code a bit to separate out the signing and file writing bits.

We should also think about where the struct normalization is happening, and try to centralize it if possible (e.g. why are we removing SignatureAnnotation here, but not any of the server generated fields)?

Reply:
> In sign cmd: it is marshal-> unmarshal, and for testing since we generate the object directly we only do unmarshal once. But separating them also makes sense..

I can see that - I think I might have been getting mixed up with the signing library vs the signing cli here. SGTM to leave as-is.

> yes, previously I added it in the copyTaskRun() but you suggest we rename the func or split out the signature extraction. Maybe I should add it in the func and rename it?

Yeah that sounds reasonable. That comment came from seeing a mismatch of the name from the actual behavior - e.g. we're saying something is a copy, but it's actually removing certain fields, etc. On the surface the behavior seems unexpected without the additional context.